.github/workflows/ci.yml

```diff
@@ -0,0 +1,144 @@
+      GITHUB_API_TOKEN: ${{ secrets.GITCOIN_API_TOKEN }}
+      POLYGONSCAN_API_KEY: ${{ secrets.POLYGONSCAN_API_KEY }}
+
+    # steps to perform in job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "yarn"
+
+      - name: Use Python ${{ matrix.python-version }}
+        uses: "actions/setup-python@v2"
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+
+      - name: Setup Env
+        run: |
+          echo "PYTHONPATH=/home/runner/work/web/web/app" >> $GITHUB_ENV
+          cp app/app/ci.env app/app/.env
+          pip install pip==20.0.2 setuptools wheel --upgrade
+
+      - name: Fetch and Install GeoIP database files
+        run: |
+          sudo apt-get update && sudo apt-get install -y libmaxminddb-dev libsodium-dev libsecp256k1-dev
+          cp dist/*.gz ./
+          gunzip GeoLite2-City.mmdb.tar.gz && gunzip GeoLite2-Country.mmdb.tar.gz
+          tar -xvf GeoLite2-City.mmdb.tar && tar -xvf GeoLite2-Country.mmdb.tar
+          sudo mkdir -p /opt/GeoIP/
+          sudo mv GeoLite2-City_20200128/*.mmdb /opt/GeoIP/
+          sudo mv GeoLite2-Country_20200128/*.mmdb /opt/GeoIP/
+
+      - name: Install libvips, Node, and Python dependencies
+        run: |
+          sudo apt-get install -y libvips libvips-dev
+          node --version
+          yarn install
+          pip install -r requirements/test.txt
+          yarn run eslint
+          yarn run stylelint
+          (cd app; python ./manage.py collectstatic --noinput --disable-collectfast)
+
+      - name: Run management commands
+        run: |
+          python app/manage.py migrate
+          python app/manage.py fetch_gas_prices
+
+      - name: Run Python and UI tests
+        run: |
+          pytest -p no:ethereum -p no:warnings
+          bin/ci/cypress-run
+
+      - name: Generate Markdown documentation and static docs page
+        run: pydocmd build
+
+      - name: Deploy to Github Pages 🚀
+        uses: peaceiris/actions-gh-pages@v3
+        if: github.ref == 'refs/heads/master'
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: _build/site
+          cname: docs.gitcoin.co
+
+      - name: Deploy master to Docker Hub 🚀
+        if: github.ref == 'refs/heads/master' && github.event_name != 'pull_request'
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+          docker pull python:3.6-slim-jessie
+          docker build -t gitcoin/web:${{ github.run_number }} .
+          docker push gitcoin/web:${{ github.run_number }}
+          docker tag gitcoin/web:${{ github.run_number }} gitcoin/web:latest
+          docker push gitcoin/web:latest
+
+      - name: Deploy stable to Docker Hub 🚀
```

Comment:
why do have the dockerhub in ci for the stable  ?

Reply:
'twas present in our travis ci setup... a possible use i can think of will be the need to pull the stable image when we have a dockerized prod environment

Reply:
Yeah, I'm not sure we're using this atm, no harm in leaving it in place for now but it is a unnecessary step (how long does this take? Will we save compute by skipping it?).

Reply:
@gdixon deploy of master to dockerhub took around 13 minutes (seems like a lot)

it could be commented out now until we need it

Reply:
I think that would be a sane approach here @chibie - lets give it a few weeks and if we don't plan on deploying the image then we can drop it altogether, but that leg of the path-to-production is still up for debate!